File: repositories/dlq_repository.go
package repositories

import (
	"context"

	"doctor-service/entities"

	"github.com/google/uuid"
	"gorm.io/gorm"
)

type DLQRepository interface {
	Create(ctx context.Context, deadJob *entities.VerificationDeadJob) error
	FindAll(ctx context.Context) ([]entities.VerificationDeadJob, error)
	FindByID(ctx context.Context, id uuid.UUID) (*entities.VerificationDeadJob, error)
	Delete(ctx context.Context, id uuid.UUID) error
}

type GormDLQRepository struct {
	db *gorm.DB
}

func NewDLQRepository(db *gorm.DB) DLQRepository {
	return &GormDLQRepository{db: db}
}

func (r *GormDLQRepository) Create(ctx context.Context, deadJob *entities.VerificationDeadJob) error {
	return r.db.WithContext(ctx).Create(deadJob).Error
}

func (r *GormDLQRepository) FindAll(ctx context.Context) ([]entities.VerificationDeadJob, error) {
	var jobs []entities.VerificationDeadJob
	err := r.db.WithContext(ctx).Order("failed_at DESC").Find(&jobs).Error
	return jobs, err
}

func (r *GormDLQRepository) FindByID(ctx context.Context, id uuid.UUID) (*entities.VerificationDeadJob, error) {
	var job entities.VerificationDeadJob
	err := r.db.WithContext(ctx).Where("id = ?", id).First(&job).Error
	return &job, err
}

func (r *GormDLQRepository) Delete(ctx context.Context, id uuid.UUID) error {
	return r.db.WithContext(ctx).Where("id = ?", id).Delete(&entities.VerificationDeadJob{}).Error
}

File: backend_api/repositories/dlq_repository.py
from typing import List, Optional
from uuid import UUID
from sqlalchemy.orm import Session
from entities.dlq import VerificationDeadJob

class DLQRepository:
    def __init__(self, db: Session):
        self.db = db

    def Create(self, dead_job: VerificationDeadJob) -> None:
        self.db.add(dead_job)
        self.db.commit()
        self.db.refresh(dead_job)

    def FindAll(self) -> List[VerificationDeadJob]:
        return self.db.query(VerificationDeadJob).order_by(VerificationDeadJob.failed_at.desc()).all()

    def FindByID(self, id: UUID) -> Optional[VerificationDeadJob]:
        return self.db.query(VerificationDeadJob).filter(VerificationDeadJob.id == id).first()

    def Delete(self, id: UUID) -> None:
        self.db.query(VerificationDeadJob).filter(VerificationDeadJob.id == id).delete()
        self.db.commit()
